### ISSUE TITLE
Clarify specification of p:for-each

### ISSUE THREAD

**Issue by xml-project** [xproc/Vnext#37]
The current specification [says](https://spec.xproc.org/3.0/xproc/#p.for-each):

> The p:for-each will apply that subpipeline to each document in the sequence in turn.

The phrasing "in turn" might suggest, that `p:for-each` has to be implemented iterative, so that its guaranted, that the subpipeline with document 1 is excecuted before document 2's subpipeline.
The specification for [xsl:for-each](https://www.w3.org/TR/xslt-30/#for-each) does not suggest this restriction:

> The [xsl:for-each](https://www.w3.org/TR/xslt-30/#element-for-each) instruction processes each item in a sequence of items, evaluating the [sequence constructor](https://www.w3.org/TR/xslt-30/#dt-sequence-constructor) within the [xsl:for-each](https://www.w3.org/TR/xslt-30/#element-for-each) instruction once for each item in that sequence.

I wonder whether it would be better to align with XSLT's concept.

**Comment by gimsieke** [comment]
I agree.

**Comment by ndw** [comment]
Yes. I don't think the phrase "in turn" is intended to suggest that strictly sequential processing is required.
